app/controllers/users_controller.rb

```diff
@@ -21,9 +22,54 @@ def create
     end
   end
 
-  private
+  def show
+  end
+
+  def edit
+  end
 
+  def update
+    if @user.update user_params
+      flash[:success] = t ".profile_updated"
+      redirect_to @user
+    else
+      render :edit
+    end
+  end
+
+  def destroy
+    if @user.destroy
+      flash[:success] = t ".user_deleted"
+    else
+      flash[:error] = t ".deleted_failed"
+    end
+    redirect_to users_url
+  end
+
+  private
   def user_params
     params.require(:user).permit :name, :email, :password,:password_confirmation
   end
+
+  def logged_in_user
+    return if logged_in?
+    store_location
+    flash[:danger] = t ".please_log_in"
+    redirect_to login_url
+  end
+
+  def correct_user
+    redirect_to(root_url) unless current_user.current_user? @user
+  end
+
+  def admin_user
+    redirect_to(root_url) unless current_user.admin?
+  end
+
+  def load_user
+    @user = User.find(params[:id])
```

Comment:
nên dùng find_by nhé
```suggestion
    @user = User.find_by id: params[:id]
```